### DISCUSSION TITLE
Please provide a way to add extra information to commit messages

### DISCUSSION THREAD

**Discussion by SpoonMeiser** [renovatebot/renovate#22312] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

We've recently changes some of our git projects to require a reference to a ticket tracking system, by way of a regex on commit messages.

Obviously renovate isn't working on a ticket, so for this case, we want the commit message to include a special token, and our regex will check that the commit message either has something that looks like a ticket id, or this special token, and we can be sure that the only times commits don't reference are on purpose (ie. they're not missing just because a developer forgot)

The `commitMessage` configuration option is deprecated, so we shouldn't use that, as it'll break in the future.

`commitMessagePrefix` is used to add semantic commit labels, which we need (and we rely on them being chore in some cases and fix in others, so we can't hard code the semantic label and our token in here)
`commitMessageSuffix` is sometimes used to convey additional information, so we can't use this without losing that

In fact, there seems to be no way to add this additional token into commit messages apart from setting `commitMessage`.

Could a mechanism be added to allow us to add this token somewhere? Perhaps add another config option that is used to make up the commit message, but is explicitly not used by renovate internally?

Or is there a way to include both the semantic label and a hard-coded token into `commitMessagePrefix`? If so, this needs to be documented.

**Comment by rarkins** [comment]
Does it need to be in the first line of the commit message? If not then [commitBody](https://docs.renovatebot.com/configuration-options/#commitbody) could help you

**Comment by SpoonMeiser** [comment]
We're using gitlab's commit message regex to enforce this, it would be difficult to switch to something else.

Actually, one of my colleagues found a solution to this:
```
"commitMessagePrefix": "{{{semanticCommitType}}}({{{semanticCommitScope}}}): NON-JIRA"
```

Where "NON-JIRA" is the token we want to add.

The thing is, you can't figure this out from the documentation, you have to make some pretty big assumptions and take a leap of faith.

If this is an acceptable way of doing this - there's no unforeseen problems we're likely to run into - then it might be a good idea to add some more details to the documentation about what is available in template strings, and what config options you can use templates for.

**Reply by rarkins** [reply]
Docs PR welcomed if you find it works - sounds like it could live in the `commitMessagePrefix` section.
